module/migration.mjs

```diff
@@ -30,11 +32,14 @@ export const migrateWorld = async function() {
     .concat(Array.from(game.items.invalidDocumentIds).map(id => [game.items.getInvalid(id), false]));
   for ( const [item, valid] of items ) {
     try {
+      const flags = { needsMigration: false };
       const source = valid ? item.toObject() : game.data.items.find(i => i._id === item.id);
-      const updateData = migrateItemData(source, migrationData);
-      if ( !foundry.utils.isEmpty(updateData) ) {
+      const updateData = migrateItemData(source, migrationData, flags);
+      if ( !foundry.utils.isEmpty(updateData) || flags.needsMigration ) {
         console.log(`Migrating Item document ${item.name}`);
-        await item.update(updateData, {enforceTypes: false, diff: valid});
+        const update = foundry.utils.isEmpty(updateData) ? source : updateData;
+        update["flags.dnd5e.-=needsMigration"] = null;
```

Comment:
Maybe it would be a little neater if this was unset at the point `flags.needsMigration` is set? So if we amend `migrateItemData` to:

```js
if ( foundry.utils.getProperty(item, "flags.dnd5e.needsMigration") ) {
  flags.needsMigration = true;
  updateData["flags.dnd5e.-=needsMigration"] = null;
}
```

Then perhaps we can avoid having to unset it elsewhere? Additionally it means that `updateData` will never be empty if `flags.needsMigration` is true, so we can simplify the `if` guards somewhat.